### ISSUE TITLE
Incorrect Header Check since 6.15

### ISSUE THREAD

**Issue by kyralretsam** [Soreepeong/XivMitmLatencyMitigator#47]
I have been running this on my RPi3 for about a week with no issues (MCH IS HAPPY!) but this afternoon when I went to login following the 6.15 patch, it wouldn't load. I set my PS4 back to normal networking and it works fine. Here is the output from the script via journalctl

```
pi@pihole:~ $ sudo systemctl status xivalex
* xivalex.service - XIV Alexander Daemon
     Loaded: loaded (/etc/systemd/system/xivalex.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-06-07 16:24:54 EDT; 1min 57s ago
   Main PID: 4606 (XIVAlexanderMIT)
      Tasks: 4 (limit: 2060)
     Memory: 11.9M
        CPU: 871ms
     CGroup: /system.slice/xivalex.service
             |-4606 /bin/bash - /usr/local/bin/XIVAlexanderMITM.sh
             |-4608 /usr/bin/python3 /usr/local/bin/XIVAlexanderMITM.py
             |-4783 /usr/bin/python3 /usr/local/bin/XIVAlexanderMITM.py
             `-4784 /usr/bin/python3 /usr/local/bin/XIVAlexanderMITM.py

Jun 07 16:26:42 pihole XIVAlexanderMITM.sh[4784]:     bundle_data = bytearray(zlib.decompress(bundle_data))
Jun 07 16:26:42 pihole XIVAlexanderMITM.sh[4784]: zlib.error: Error -3 while decompressing data: incorrect header check
Jun 07 16:26:42 pihole XIVAlexanderMITM.sh[4784]: 2022-06-07 16:26:42,913        4784        INFO        D->U discarded 00
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4608]: 2022-06-07 16:26:43,022        4608(main)        INFO        [4732  ] has exit with status code 0.
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4783]: 2022-06-07 16:26:43,288        4783        ERROR        Unknown error occurred while trying to parse bundle
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4783]: Traceback (most recent call last):
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4783]:   File "/usr/local/bin/XIVAlexanderMITM.py", line 452, in find
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4783]:     bundle_data = bytearray(zlib.decompress(bundle_data))
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4783]: zlib.error: Error -3 while decompressing data: incorrect header check
Jun 07 16:26:43 pihole XIVAlexanderMITM.sh[4783]: 2022-06-07 16:26:43,293        4783        INFO        D->U discarded 00
```

**Comment by sleong67** [comment]
Hi. I'm having this exact same issue. It began to happen today.

![image](https://user-images.githubusercontent.com/81522184/172510155-ed8ecaaf-a90a-4ac0-8e22-ec7edd4bd3b7.png)

Any updates on this issue?

**Comment by bankjaneo** [comment]
I also have the same problem. I think **6.15** breaks XivMitmLatencyMitigator even with updated Opcode. Thank you dev in advance for the fix he's working on now.

**Comment by RayLynx** [comment]
![unknown-11](https://user-images.githubusercontent.com/98320763/172695358-586d7e98-6cb6-49ab-9cec-173ec38728f5.png)
Today early

![unknown-11(1)](https://user-images.githubusercontent.com/98320763/172695390-d769ba09-0c07-43e2-afcf-336c9df358fd.png)
At the moment

**Comment by RayLynx** [comment]
![unknown-29(1)](https://user-images.githubusercontent.com/98320763/172695581-5ff0fa82-d61c-40a7-a494-a88d8a85143c.png)
Today early
![Uploading unknown-11(1).png…]()
At the moment

Well... Thanks SE

**Comment by RayLynx** [comment]
![unknown-11(1)](https://user-images.githubusercontent.com/98320763/172695946-478290a3-19b9-47d5-80e0-8cb7e5ef6f3c.png)

**Comment by kyralretsam** [comment]
@RayLynx Doesn't look like a problem with the script, you just can't hit github. Have you verified network connectivity? (ie, can you hit sites like Google)

**Comment by Soreepeong** [comment]
It should work now, but you need to copy ffxiv.exe from Windows/Mac installation into the linux machine, and the machine must be running under x86 or x86_64. For support for other platforms, I'm accepting contributions. This script is pretty lightweight though, so if you run a rpi or something, a small x86 vm under qemu might work.

**Comment by kyralretsam** [comment]
Question, I see in the commit that you reference the HFS file server. Is that just to get the ffxiv.exe from your Windows/Mac machine to the VM/Linux box?

**Comment by Soreepeong** [comment]
Yeah, if you know how to use `scp`, then you're good to go!

**Comment by aurilain** [comment]
Any idea how to fix this?

![image](https://user-images.githubusercontent.com/71463763/172756359-a0b08fd2-5b01-4b71-9b77-6b53f5515d7c.png)

**Comment by Soreepeong** [comment]
Try `-o` instead. I keep confusing between small o and big o there

**Comment by aurilain** [comment]
That  started to work, however now it appears to not actually be downloading anything:

![image](https://user-images.githubusercontent.com/71463763/172757137-4c3f2ef0-cf53-43d2-88c5-d65c63fe162a.png)

**Comment by Soreepeong** [comment]
That would mean that it can't connect to HFS. Is that IP visible to the VM? You may have to pick different local IP.

**Comment by bankjaneo** [comment]
> It should work now, but you need to copy ffxiv.exe from Windows/Mac installation into the linux machine, and the machine must be running under x86 or x86_64. For support for other platforms, I'm accepting contributions. This script is pretty lightweight though, so if you run a rpi or something, a small x86 vm under qemu might work.

Got it working now. Thank you very much. Do I need to copy update ffxiv.exe to the VM after new patch? My guess is yes. But that's not a problem :)

**Comment by aurilain** [comment]
Ok, changing the of HFS worked, and I was able to get the file into hyper-v.

However, after trying to run the mitigate.py, I get this:

![image](https://user-images.githubusercontent.com/71463763/172759204-d6dea1ad-8fdf-42f2-b34c-2f05c53ccd6e.png)

Thanks for all your help, im pretty tech illiterate

**Comment by bankjaneo** [comment]
> Ok, changing the of HFS worked, and I was able to get the file into hyper-v.
> 
> However, after trying to run the mitigate.py, I get this:
> 
> ![image](https://user-images.githubusercontent.com/71463763/172759204-d6dea1ad-8fdf-42f2-b34c-2f05c53ccd6e.png)
> 
> Thanks for all your help, im pretty tech illiterate

For Ubuntu I fixed with this command.

`sudo apt install g++ gcc-multilib g++-multilib`

**Comment by Soreepeong** [comment]
@bankjaneo You probably won't have to do that; in fact, this code only works with 6.15 ffxiv.exe at the moment.

@aurilain Try typing `pacman -S gcc` and after it's done, try it again.

**Comment by aurilain** [comment]
I tried this series of commands:

![image](https://user-images.githubusercontent.com/71463763/172759859-c9e65770-e136-41e6-8190-6c7be67205db.png)

Do i need to change some settings to have actual disk space in the VM?

**Comment by Soreepeong** [comment]
That looks like archiso's default ramdisk does not have enough space to accommodate those installs. 

This may work `mount -o remount,size=4G tmpfs /` , but if it doesn't, I'll have to think of something else.

**Comment by aurilain** [comment]
Sadly, that didn't do anything. I'll troubleshoot this more later, thanks for all your help again

**Comment by Arkzera** [comment]
Hi, im having the exact samee problems as @aurilain, is installing the vm the only way to make it work?

**Comment by Arkzera** [comment]
Quick update, I had to install an ubuntu and run `sudo apt install g++ gcc-multilib g++-multilib`, its working now

**Comment by brwjon** [comment]
![Capture5](https://user-images.githubusercontent.com/88693665/172834823-09c89eb1-7f4a-49f9-85e5-b17af840f2e3.PNG)

Similar issue, had to actually install the vm. Now I get this, what am I doing wrong?

**Comment by bankjaneo** [comment]
@brwjon Did you install gcc yet?

Ubuntu
````
sudo apt install g++ g++-multilib
````

Arch
````
pacman -S gcc
````

**Comment by brwjon** [comment]
@bankjaneo Yes I did

**Comment by RayLynx** [comment]
![Sem título](https://user-images.githubusercontent.com/98320763/172850221-b2107a14-7d52-4372-aebd-d085aa8d5864.png)
when do pacman -S gcc get this error

**Comment by bankjaneo** [comment]
@RayLynx Can you try `pacman -Sy gcc`

**Comment by RayLynx** [comment]
Not enough free disk space

**Comment by brwjon** [comment]
> Edited: Change pacman parameter to -Sy

@bankjaneo yep, gcc installed, even tried reinstalling it. still getting the same error when running mitigate.py

**Comment by bankjaneo** [comment]
@RayLynx Try check disk space by `df -h` If it's not enough then you need to increase it first.

@brwjon Then I guess, from your picture, there should be something wrong with permission. If you are on Arch, I think you might need to wait for someone who can get it run on Arch to guide you. Personally, I'm on Ubuntu 22.04 and it's working fine now.

**Comment by brwjon** [comment]
So I put mitigate.py on my VM, set permissions to full, and I'm now getting a "failed to compile helper" error

![gcc](https://user-images.githubusercontent.com/88693665/172876365-c577901c-b5b8-476d-8cf7-3e910739f008.PNG)

**Comment by RayLynx** [comment]
i think i will have to install OS, tried change the hyper v virtual disk space but did not work

**Comment by Mayushii-chan** [comment]
Tried an installation of Arch Linux and Ubuntu, but getting the same error in both installations that @brwjon had.

**Comment by bankjaneo** [comment]
@Mayushii-chan @brwjon Out of curiosity, the ffxiv.exe file that you are using is the original one or modified one with XivAlexander?

**Comment by RayLynx** [comment]
@bankjaneo  you re running ubuntu on VM??

**Comment by bankjaneo** [comment]
@RayLynx Yes, I'm running Ubuntu 22.04 on VM. However, the version shouldn't matter.

**Comment by RayLynx** [comment]
@bankjaneo  what VM you re running it? you think if i install ubuntu system on my pc can make it works?

**Comment by brwjon** [comment]
@bankjaneo The .exe that I patched with Alexander is the dx11 one, and I have the dx9 one in my VM so I don't think it changes much

@Mayushii-chan I figured it out, got it working. I downloaded mitigate.py and copied it into my VM, then set permissions to read write & execute for the file and the folder it's in. That gets past the "permission denied" error. The other problem I had was a mismatch of gcc libraries, but while installing gcc there is a little bit of text that mentions the optional 32 bit libraries that can be downloaded. That fixed that problem. It's working fine now o7

**Comment by bankjaneo** [comment]
@RayLynx Actually I'm utilizing my VPS that hosted on Amazon EC2 to act as both VPN and XivMitmLatencyMitigator. Technically, it's just another VM that hosted outside my home. 

If you want to run this at home, any Linux VM running on QEMU, VirtualBox or Hyper-V should be fine as long as you are not running it on the same pc that you are using to play FFXIV.

What is your current setup anyway?

@brwjon Congratz. And you are right, ffxiv.exe (dx9) didn't get patched. It should be fine.

**Comment by RayLynx** [comment]
windows 10 
8gb ram i5
820m

**Comment by Mayushii-chan** [comment]
@bankjaneo A friend sent it to me, so I can only assume it's original. Definitely not modified by XivAlexander, though.

@brwjon Thank you! I was actually able to get around it by running "sudo su" before the mitigate.py command. Probably a far worse and less elegant solution to it than yours, but at least it is working.

**Comment by RayLynx** [comment]
![P20609-162312](https://user-images.githubusercontent.com/98320763/172928144-0bd1cc0a-027b-4187-bd4c-a05395cfc80b.jpg)
With ubuntu

**Comment by Arkzera** [comment]
> ![P20609-162312](https://user-images.githubusercontent.com/98320763/172928144-0bd1cc0a-027b-4187-bd4c-a05395cfc80b.jpg) With ubuntu

Do `sudo apt install g++ gcc-multilib g++-multilib 
U need to import the ffxiv.exe into your vm too, use HFS to do it, the link and a quick tutorial is on readme.

**Comment by RayLynx** [comment]
ohhhh i thought i just need throw the ffxiv.exe on VM directory

**Comment by brwjon** [comment]
> 

I set up the disk with UEFI, then section 4 and 5. Just basically to get it to a bootable state. 

for the gcc stuff:

pacman -S gcc

After that, you need to install the 32bit libraries, when installing, there's a bit of text that says you can optionally install them. just use the same command with that, instead of gcc

**Comment by brwjon** [comment]
> @brwjon from the video you sent me, what steps did you take to install the OS? Do you have a command to install the specifically needed gcc file?

I set up the disk with UEFI, then section 4 and 5. Just basically to get it to a bootable state.

for the gcc stuff:

pacman -S gcc

After that, you need to install the 32bit libraries, when installing, there's a bit of text that says you can optionally install them. just use the same command with that, instead of gcc

**Comment by DiegoSalazar231** [comment]
Hi. I'm having troubles with the step 7, I'm not sure how to continue since I wrote the same ip from HFS, but somehow doesn't work. Sorry if the answer is too obvious, I don't have experience on these programs. 

![20220609_174159](https://user-images.githubusercontent.com/98240637/172957662-8029f0d5-c922-416e-893c-02ad74222526.jpg)

**Comment by brwjon** [comment]
@DiegoSalazar231  in HFS, go to menu -> IP address, and pick another one of those, it should give you a new link that won't time out

**Comment by DiegoSalazar231** [comment]
@brwjon Thanks. I picked a new one, and it doesn't time out, but I didn't have to wait a single second, is that ok? I still can't download the ffxiv.exe and only says this :c 

![20220609_180211](https://user-images.githubusercontent.com/98240637/172959445-4bc42d52-1f64-40f8-8e6d-b3d3ea2296f9.jpg)

**Comment by kyralretsam** [comment]
I stood up a Debian VM on my Synology NAS and reinstalled the script there and it works perfectly :)  Thanks much!

I did make a systemd unit file and startup/cleanup scripts to go with it if you want them

**Comment by Rappy28** [comment]
> @brwjon from the video you sent me, what steps did you take to install the OS?
> Do you have a command to install the specifically needed gcc file?

You do not have to install Arch Linux to increase disk space! I got all of this to work on the simple ISO VM today.
Just hit Tab when you are booting up your VM. It's going to bring up a line of options on the bottom. Just add `cow_spacesize=1G` at the end of the line then continue the VM launch as usual. This gave me enough disk space to install gcc and its 32-bit lib.

I would also recommend using python3's httpserver command to copy ffxiv.exe onto the VM. I could not get HFS to run on OSX Catalina as it only exists as a .exe. I regret spending hours on trying to get Wine to work on Catalina while I could have just used python...

**Comment by sleong67** [comment]
Hi guys. 

I did all the mentioned steps, even increased / space with cow_spacesize=1G, but when I tried to run the program is giving me this error. I think it might have to do something with the gcc libraries.

![image](https://user-images.githubusercontent.com/81522184/173479761-e92ba3fe-52b3-4d55-9a70-a0f745c3c5af.png)


Thanks in advance.

**Comment by madara4567** [comment]
syntax errors, permission denied, and now g++ isnt taking the Soreepeong curl ... What the fuck is going on here. Tried 4 linux distributions, fully installed. Nothing works.
